### PR TITLE
create trieStorageManagerWithoutCheckpoints

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
Add a new flag "CheckpointsEnabled" that will create a new trieStorageManagerWithoutCheckpoints if it is set to false

### CODE REVIEW

File: data/interface.go [hunk 1]
```diff
@@ -225,7 +225,7 @@ type StorageManager interface {
 
 // TrieFactory creates new tries
 type TrieFactory interface {
-	Create(config.StorageConfig, string, bool, uint) (StorageManager, Trie, error)
+	Create(config.StorageConfig, string, bool, bool, uint) (StorageManager, Trie, error)
```
Comment (iulianpascalau) [inline data/interface.go:228]: let's switch to argument parameter here :)

File: data/trie/hashesHolder/disabled/disabledCheckpointHashesHolder.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+type disabledCheckpointHashesHolder struct {
+}
+
+// NewDisabledCheckpointHashesHolder creates a new instance of disabledCheckpointHashesHolder
+func NewDisabledCheckpointHashesHolder() *disabledCheckpointHashesHolder {
+	return &disabledCheckpointHashesHolder{}
+}
+
+// Put does nothing for this implementation
```
Comment (iulianpascalau) [inline data/trie/hashesHolder/disabled/disabledCheckpointHashesHolder.go:13]: returns false

File: data/trie/trieStorageManagerWithoutCheckpoints.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	return &trieStorageManagerWithoutCheckpoints{tsm}, nil
+}
+
+// SetCheckpoint does nothing if pruning is disabled
+func (tsm *trieStorageManagerWithoutCheckpoints) SetCheckpoint(_ []byte, _ chan core.KeyValueHolder) {
+	log.Debug("trieStorageManagerWithoutCheckpoints - SetCheckpoint is disabled")
+}
+
+// AddDirtyCheckpointHashes does nothing for this implementation
```
Comment (iulianpascalau) [inline data/trie/trieStorageManagerWithoutCheckpoints.go:30]: returns false

File: data/interface.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply data/interface.go:228]: Done.

File: data/trie/hashesHolder/disabled/disabledCheckpointHashesHolder.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+// RemoveCommitted does nothing for this implementation
+func (d *disabledCheckpointHashesHolder) RemoveCommitted(_ []byte) {
+}
+
+// Remove does nothing for this implementation
+func (d *disabledCheckpointHashesHolder) Remove(_ []byte) {
+}
+
+// ShouldCommit does nothing for this implementation
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply data/trie/hashesHolder/disabled/disabledCheckpointHashesHolder.go:26]: Done.

File: data/trie/trieStorageManagerWithoutCheckpoints.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply data/trie/trieStorageManagerWithoutCheckpoints.go:30]: Done.

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — DISMISSED** [review]

File: cmd/node/config/config.toml [hunk 5]
```diff
@@ -607,6 +607,7 @@
 
 [StateTriesConfig]
     CheckpointRoundsModulus = 100
+    CheckpointsEnabled = true
```
Comment (sasurobert) [inline cmd/node/config/config.toml:610]: is the default true ? with checkpoints ?

**Review by sasurobert — DISMISSED** [review]

File: cmd/node/config/config.toml (hunk 5 above)
Comment (iulianpascalau) [reply cmd/node/config/config.toml:610]: Will discuss and will eventually change on the testnet/devnet/mainnet if this should be set to false.

**Review by sasurobert — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**